src/Repository/CarrinhoRepository.php

```diff
@@ -40,4 +40,11 @@ public function __construct(ManagerRegistry $registry)
     //            ->getOneOrNullResult()
     //        ;
     //    }
+
+    public function salvarCarrinho(Carrinho $carrinho) : void
```

Comment:
o repository já é de Carrinho, deixa apenas salvar...
aproveita e remover o comentário acima

```suggestion
    public function salvar(Carrinho $carrinho) : void
```